File: firstTask.cpp
#include <iostream>
#include <string>
#include <vector>

int main() {
    // Создаем строку, аналогичную str_1 = 'AaBbCcDd'
    std::string str_1 = "AaBbCcDd";
    
    // Выводим символы с шагом 2, начиная с первого
    std::cout << "str_1[::2] -> ";
    for(size_t i = 0; i < str_1.size(); i += 2) {
        std::cout << str_1[i];
    }
    std::cout << std::endl;
    
    // Выводим символы с шагом 2, начиная со второго
    std::cout << "str_1[1::2] -> ";
    for(size_t i = 1; i < str_1.size(); i += 2) {
        std::cout << str_1[i];
    }
    std::cout << std::endl;
    
    // Создаем вектор, аналогичный li = ['a','1','b','2','c','3']
    std::vector<std::string> li = {"a", "1", "b", "2", "c", "3"};
    
    // Создаем два новых вектора
    std::vector<std::string> li_1;
    std::vector<std::string> li_2;
    
    // Заполняем li_1 элементами с четными индексами
    for(size_t i = 0; i < li.size(); i += 2) {
        li_1.push_back(li[i]);
    }
    
    // Заполняем li_2 элементами с нечетными индексами
    for(size_t i = 1; i < li.size(); i += 2) {
        li_2.push_back(li[i]);
    }
    
    // Освобождаем память, аналогичное del li
    li.clear();
    
    // Выводим результаты
    for(const auto& item : li_1) {
        std::cout << item << " ";
    }
    std::cout << std::endl;
    
    for(const auto& item : li_2) {
        std::cout << item << " ";
    }
    std::cout << std::endl;
    
    return 0;
}

File: firstTask.java
public class Main {
    public static void main(String[] args) {
        // Работа со строкой
        String str_1 = "AaBbCcDd";
        
        // Эквивалент str_1[::2] - получаем символы с шагом 2, начиная с 0
        String result1 = getSubstring(str_1, 0, 2);
        System.out.println("str_1[::2] -> " + result1);
        
        // Эквивалент str_1[1::2] - получаем символы с шагом 2, начиная с 1
        String result2 = getSubstring(str_1, 1, 2);
        System.out.println("str_1[1::2] -> " + result2);
        
        // Работа со списком
        String[] li = {"a", "1", "b", "2", "c", "3"};
        
        // Создаем два новых массива
        String[] li_1 = getElements(li, 0, 2);
        String[] li_2 = getElements(li, 1, 2);
        
        // Эквивалент del li - очищаем массив
        li = null;
        
        // Выводим результаты
        printArray(li_1);
        printArray(li_2);
    }
    
    // Метод для получения подстроки с заданным шагом
    private static String getSubstring(String str, int start, int step) {
        StringBuilder result = new StringBuilder();
        for (int i = start; i < str.length(); i += step) {
            result.append(str.charAt(i));
        }
        return result.toString();
    }
    
    // Метод для получения элементов массива с заданным шагом
    private static String[] getElements(String[] arr, int start, int step) {
        List<String> temp = new ArrayList<>();
        for (int i = start; i < arr.length; i += step) {
            temp.add(arr[i]);
        }
        return temp.toArray(new String[0]);
    }
    
    // Метод для вывода массива
    private static void printArray(String[] arr) {
        for (String s : arr) {
            System.out.print(s + " ");
        }
        System.out.println();
    }
}
